packages/button/src/react/index.tsx

```diff
@@ -0,0 +1,237 @@
+    ),
+  text: () => css(stylesheet[`.psds-button__text`])
+}
+
+const mapIconSize = (size: string) => {
+  const btnToIconSizes = {
+    [vars.sizes.xSmall]: iconSizes.small,
+    [vars.sizes.small]: iconSizes.medium,
+    [vars.sizes.medium]: iconSizes.medium,
+    [vars.sizes.large]: iconSizes.medium
+  }
+  return btnToIconSizes[size] ? btnToIconSizes[size] : iconSizes.medium
+}
+
+interface RenderIconProps extends HTMLAttributes<HTMLDivElement> {
+  loading: boolean
+  icon: React.ReactNode
+  appearance: string
+  themeName: string
+  size: string
+  iconOnly: boolean
+  isLoadingWithNoText: boolean
+  iconAlign: string
+}
+
+const renderIcon: React.FC<RenderIconProps> = props =>
+  props.loading ? (
+    <div
+      {...styles.icon({
+        iconAlign: props.iconAlign,
+        iconOnly: props.iconOnly,
+        isLoadingWithNoText: props.isLoadingWithNoText
+      })}
+    >
+      <Icon size={mapIconSize(props.size)}>
+        <span
+          {...styles.loading({
+            appearance: props.appearance,
+            themeName: props.themeName
+          })}
+        />
+      </Icon>
+    </div>
+  ) : props.icon ? (
+    <div
+      {...styles.icon({
+        iconAlign: props.iconAlign,
+        iconOnly: props.iconOnly,
+        isLoadingWithNoText: props.isLoadingWithNoText
+      })}
+    >
+      {React.cloneElement(props.icon as React.ReactElement, {
+        size: mapIconSize(props.size)
+      })}
+    </div>
+  ) : null
+
+renderIcon.propTypes = {
+  loading: PropTypes.bool,
+  icon: PropTypes.element
+}
+
+interface ButtonStatics {
+  appearances: typeof vars.appearances
+  iconAligns: typeof vars.iconAligns
+  sizes: typeof vars.sizes
+}
+
+interface ButtonProps extends HTMLAttributes<HTMLElement> {
+  appearance?: string
+  disabled?: boolean
+  href?: string
+  icon?: React.ReactNode
+  iconAlign?: string
+  loading?: boolean
+  size?: string
+}
+
+interface ButtonComponent
+  extends RefForwardingComponent<ButtonProps, HTMLElement, ButtonStatics> {}
+
+const Button = React.forwardRef<HTMLElement, ButtonProps>(
+  (props, forwardedRef) => {
+    const themeName = useTheme()
+    const ref = React.useRef<HTMLAnchorElement | HTMLButtonElement>()
+    React.useImperativeHandle(forwardedRef, () => ref.current)
+    const nonLoadingWidth = React.useMemo(() => {
+      if (props.loading && ref && ref.current) {
+        return ref.current.offsetWidth
+      }
+    }, [props.loading, ref])
+
+    const tagName = props.href ? 'a' : 'button'
+    const isLoadingWithNoText = !!nonLoadingWidth
+    const allProps = {
+      ...props,
+      isLoadingWithNoText,
+      iconOnly: React.Children.count(props.children) <= 0,
+      themeName
+    }
+
+    const isDisabledLink = allProps.disabled && allProps.href
+    const filteredProps = filterReactProps(props, { tagName })
+    delete filteredProps.icon
+    if (isDisabledLink) {
+      delete filteredProps.onClick
+    }
+
+    return React.createElement(
```

Comment:
i wonder why we were using createElement....guessing this was one of the first components and maybe we didn't support jsx then. just thinking out load - don't change anything for this pr but i might follow up with a conversion to jsx...